call.go

```diff
@@ -30,6 +30,22 @@ type CallOption interface {
 	applyToCall(*callCfg)
 }
 
+type requestHeadersOption struct{}
+
+func (requestHeadersOption) applyToCall(*callCfg) {}
+
+func RequestHeaders(MutableHeader) CallOption {
```

Comment:
I've been following `net/http`, which calls the `map[string][]string` type `Header` (note the singular). I think we should be consistent, so these options would be `RequestHeader` and `ResponseHeader`. I'm also fine with changing the other types (Mutable/Immutable/etc) to be plural.